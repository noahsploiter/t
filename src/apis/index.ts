export * from './getMainPage'
export * from './login'
export * from './logout'
export * from './route'
