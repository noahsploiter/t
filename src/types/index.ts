export * from './Counting'
export * from './LowerLetter'
export * from './Paging'
export * from './SearchOptions'
export * from './SearchOrdering'
export * from './SearchPeriod'
export * from './Segment'
export * from './SexualOrientation'
export * from './ThumbSize'
export * from './VideoDetail'
export * from './VideoResponse'
