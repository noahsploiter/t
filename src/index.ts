import { Route, getMainPage, login, logout } from './apis'
import { getAutoComplete } from './apis/autoComplete'
import { getToken } from './apis/getToken'
import { Engine } from './core/engine'
import { WebMaster } from './core/webmaster'
import { albumPage } from './scrapers/pages/album'
import { modelPage } from './scrapers/pages/model'
import { photoPage } from './scrapers/pages/photo'
import { pornstarPage } from './scrapers/pages/pornstar'
import { videoPage } from './scrapers/pages/video'
import { albumSearch } from './scrapers/search/album'
import { gifSearch } from './scrapers/search/gif'
import { modelSearch } from './scrapers/search/model'
import { pornstarSearch } from './scrapers/search/pornstar'
import { pornstarList } from './scrapers/search/pornstars'
import { videoSearch } from './scrapers/search/video'
import type { AlbumSearchOptions, AutoCompleteOptions, GifSearchOptions, PornstarListOptions, PornstarSearchOptions, VideoSearchOptions } from './types'
import type { RequestInit } from 'node-fetch'

export * from './types'
export * from './utils/error'
export type { AlbumPage } from './scrapers/pages/album'
export type { PhotoPage } from './scrapers/pages/photo'
export type { VideoPage } from './scrapers/pages/video'
export type { PornstarPage } from './scrapers/pages/pornstar'
export type { ModelPage } from './scrapers/pages/model'

export type { AlbumSearchResult } from './scrapers/search/album'
export type { PornstarSearchResult } from './scrapers/search/pornstar'
export type { GifSearchResult } from './scrapers/search/gif'
export type { VideoSearchResult } from './scrapers/search/video'

export type { WebmasterCategory } from './apis/webmaster/categories'
export type { WebmasterDeleted } from './apis/webmaster/deleted'
export type { WebmasterEmbed } from './apis/webmaster/embed'
export type { WebmasterSearch } from './apis/webmaster/search'
export type { WebmasterStarsDetailed } from './apis/webmaster/stars_detailed'
export type { WebmasterStars, WebmasterStar } from './apis/webmaster/stars'
export type { WebmasterTags } from './apis/webmaster/tags'
export type { WebmasterVideoById } from './apis/webmaster/video_by_id'
export type { WebmasterVideoIsActive } from './apis/webmaster/video_is_active'

export class PornHub {
    engine = new Engine()
    route = Route
    webMaster = new WebMaster(this.engine)

    setAgent(agent: RequestInit['agent']) {
        this.engine.request.setAgent(agent)
    }

    setHeader(key: string, value: string) {
        this.engine.request.setHeader(key, value)
    }

    setCookie(key: string, value: any) {
        this.engine.request.setCookie(key, value)
    }

    /**
     * See: https://github.com/pionxzh/Pornhub.js/issues/27
     * @deprecated This method is no longer needed.
     */
    async warmup() {
        console.warn('`warmup` has been deprecated. You can safely remove this method call. It has been handled internally.')
        // no-op
    }

    /**
     * Login with account and password.
    */
    login(account: string, password: string) {
        return login(this.engine, account, password)
    }

    /**
     * Logout from Pornhub.com.
    */
    logout() {
        return logout(this.engine)
    }

    /**
     * Get token from Pornhub.com.
     * Most of pornhub's api need this token.
     * You can cache this token to avoid frequent requests (I'm not sure about the expiration time!).
     *
     * For now, this token is only used for `autoComplete` and `searchModel`.
     * This library will automatically get token if you don't provide it.
     */
    getToken() {
        return getToken(this.engine)
    }

    /**
     * Get video information by url/ID
     * @param urlOrId Video ID or page url
    */
    async video(urlOrId: string) {
        if (!this.engine.warmedUp) {
            // make a call to the main page to get the cookies.
            // PornHub will redirect you to a corn video if you don't have a proper cookie set.
            // See issue: [#27 Video been redirected to a corn video](https://github.com/pionxzh/Pornhub.js/issues/27)\
            await getMainPage(this.engine)
            this.engine.warmedUp = true
        }
        return videoPage(this.engine, urlOrId)
    }

    /**
     * Get album information by url/ID
     * @param urlOrId Album ID or page url
    */
    album(urlOrId: string) {
        return albumPage(this.engine, urlOrId)
    }

    /**
     * Get photo information by url/ID
     * @param urlOrId Photo ID or page url
    */
    photo(urlOrId: string) {
        return photoPage(this.engine, urlOrId)
    }

    /**
     * Get pornstar information by url/ID
     * @param urlOrName Pornstar name or page url
    */
    pornstar(urlOrName: string) {
        return pornstarPage(this.engine, urlOrName)
    }

    /**
     * Get model information by url/ID
     * @param urlOrName Model name or page url
     */
    model(urlOrName: string) {
        return modelPage(this.engine, urlOrName)
    }

    /**
     * Get autocomplete result by keyword.
     */
    autoComplete(keyword: string, options: AutoCompleteOptions = {}) {
        return getAutoComplete(this.engine, keyword, options)
    }

    /**
     * Search album by keyword.
     */
    searchAlbum(keyword: string, options: AlbumSearchOptions = {}) {
        return albumSearch(this.engine, keyword, options)
    }

    /**
     * Search gif by keyword.
     */
    searchGif(keyword: string, options: GifSearchOptions = {}) {
        return gifSearch(this.engine, keyword, options)
    }

    /**
     * Search pornstar by keyword.
     */
    searchPornstar(keyword: string, options: PornstarSearchOptions = {}) {
        return pornstarSearch(this.engine, keyword, options)
    }

    /**
     * Search model by keyword.
     */
    searchModel(keyword: string, options: AutoCompleteOptions = {}) {
        return modelSearch(this.engine, keyword, options)
    }

    /**
     * Search video by keyword.
     */
    searchVideo(keyword: string, options: VideoSearchOptions = {}) {
        return videoSearch(this.engine, keyword, options)
    }

    /**
     * Get pornstar list.
     */
    pornstarList(options: PornstarListOptions = {}) {
        return pornstarList(this.engine, options)
    }
}
