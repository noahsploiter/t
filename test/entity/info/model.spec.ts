import { describe, expect, it } from 'vitest'
import { PornHub } from '../../../src/index'

const pornhub = new PornHub()

describe('Model Info', () => {
    it('# run()', async () => {
        const url = 'https://www.pornhub.com/model/luna-okko'
        const result = await pornhub.model(url)

        const {
            avatar,
            cover,
            rank,
            subscribers,
            videoViews,
            profileViews,
            videoWatched,
            uploadedVideoCount,
            taggedVideoCount,
            ...restResult
        } = result

        expect(result.name).toBe('Luna Okko')
        expect(avatar).not.to.be.empty
        expect(cover).not.toBe(0)
        expect(rank).not.toBe(0)
        expect(subscribers).not.toBe(0)
        expect(videoViews).not.toBe(0)
        expect(profileViews).not.toBe(0)
        expect(videoWatched).not.toBe(0)
        expect(uploadedVideoCount).not.toBe(0)
        expect(taggedVideoCount).toBe(0)

        expect(restResult).toMatchSnapshot()
    })
})
