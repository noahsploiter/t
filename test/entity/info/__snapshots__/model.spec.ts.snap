// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Model Info > # run() 1`] = `
{
  "about": "I travel the world with my boyfriend, bringing you with us in our adventures!",
  "awarded": false,
  "bio": "",
  "ethnicity": "Asian",
  "fakeBoobs": false,
  "featuredIn": [],
  "gender": "Female",
  "hairColor": "Black",
  "height": "5' 2\\" (157cm)",
  "interestedIn": "Guys and Girls",
  "name": "Luna Okko",
  "piercings": false,
  "premium": false,
  "relationship": "Taken",
  "socials": {
    "amazonWishList": undefined,
    "instagram": "https://www.instagram.com/lunaokko",
    "modelhub": undefined,
    "snapchat": undefined,
    "twitter": "https://www.twitter.com/lunaokko_",
    "website": "https://lunaokko.com",
  },
  "tattoos": true,
  "verified": true,
  "weight": "88lbs. (40kg)",
}
`;
